### ISSUE TITLE
Generated recommended addons not appearing on homepage of local instance.

### ISSUE THREAD

**Issue by jrbenny35** [mozilla/addons-server#13106]
Creating addons for the integration-tests using this [code](https://github.com/mozilla/addons-server/blob/master/src/olympia/landfill/serializers.py#L87) does not create recommended addons. Also navigating to the "Extensions" page on the frontend results in no extensions loading.

The api call here ```/api/v4/addons/search/?app=firefox&page_size=4&recommended=true&sort=random&type=extension&lang=en-US&wrap_outgoing_links=true``` seems to return a 200 but when testing this, the ```results``` field is empty.

The best way to test this would be:
1. Clone addons-frontend repo
2. Edit ```config/uitests-amo.js``` and add this line: ```mozillaUserId: 10968,```
3. Create a script file and add the following to it:
```
#!/usr/bin/env sh
set -ex
# create fxa email
git clone --depth 1 https://github.com/mozilla/addons-server.git
export UITEST_FXA_EMAIL=$(python ./addons-server/tests/ui/scripts/generate_fxa_email.py)
docker build -f addons-server/Dockerfile --build-arg GROUP_ID=$(id -g) --build-arg USER_ID=$(id -u) -t addons/addons-server:latest addons-server/
docker-compose -f addons-server/docker-compose.yml -f tests/frontend/docker-compose.selenium.yml -f tests/ui/docker-compose.functional-tests.yml up -d --build
docker-compose -f addons-server/docker-compose.yml -f tests/frontend/docker-compose.selenium.yml -f tests/ui/docker-compose.functional-tests.yml ps
# Make sure dependencies get updated in worker and web container
docker-compose -f addons-server/docker-compose.yml -f tests/frontend/docker-compose.selenium.yml -f tests/ui/docker-compose.functional-tests.yml exec worker make -f Makefile-docker update_deps update_assets
docker-compose -f addons-server/docker-compose.yml -f tests/frontend/docker-compose.selenium.yml -f tests/ui/docker-compose.functional-tests.yml restart worker
docker-compose -f addons-server/docker-compose.yml -f tests/frontend/docker-compose.selenium.yml -f tests/ui/docker-compose.functional-tests.yml exec web make -f Makefile-docker update_deps update_assets
docker-compose -f addons-server/docker-compose.yml -f tests/frontend/docker-compose.selenium.yml -f tests/ui/docker-compose.functional-tests.yml restart web
docker-compose -f addons-server/docker-compose.yml -f tests/frontend/docker-compose.selenium.yml -f tests/ui/docker-compose.functional-tests.yml exec web make setup-ui-tests
docker-compose -f addons-server/docker-compose.yml -f tests/frontend/docker-compose.selenium.yml -f tests/ui/docker-compose.functional-tests.yml up -d
```
4. Create a python 3 environment, and make sure you have docker + docker-compose installed
5. Run the script within the addons-frontend root.
6. Navigate to ```olympia.test``` to view the homepage.

You will also see a few other sections missing, namely ```Popular Extensions``` and ```Recommended Themes```.

**Comment by jrbenny35** [comment]
This blocking my PR here, https://github.com/mozilla/addons-frontend/pull/9018

**Comment by eviljeff** [comment]
fwiw, running `GenerateAddonsSerializer().create_generic_featured_addons()` directly in a django shell does create the 10 featured addons that show up at http://olympia.test/api/v4/hero/ and http://olympia.test/api/v4/hero/primary/  - the former api endpoint is used on the homepage now (rather than /api/v4/addons/search)

**Comment by jrbenny35** [comment]
Correct. the Hero addons seem to load fine, but the searched ones for the shelves don't. The call to ```/api/v4/addons/search/?app=firefox&page_size=4&recommended=true&sort=random&type=extension&lang=en-US&wrap_outgoing_links=true HTTP/1.1```, should return something as the addons created are recommended (if you go to their specific pages).

**Comment by eviljeff** [comment]
If the add-ons show as a recommended on their detail pages but not in the search results then it's elasticsearch indexing.  The data for the addon - and it's recommended status - is updated via a celery task so, depending on what else happens between `create_generic_featured_addons` being called and `api/v4/addons/search` the results might not be up to date.

**Comment by jrbenny35** [comment]
Fixed in #13130
